[25] 주소 수정, 삭제, 리스트

ohsuha opened the pull request:
- 배송지 조회, 수정, 삭제 api 입니다
- 리스트에서는 페이지를 0으로 받도록 했는데 1로 받는게 나은가요?
- Create 할때 reponse 는 없애려고 합니다

ohsuha committed:
[25] 주소 수정, 삭제, 리스트

- 리스트에서는 페이지를 0으로 받도록 했는데 1로 받는게 나은가요?
- Create 할때 reponse 는 없애려고 합니다

ohsuha committed:
[25] 주소 수정, 삭제, 리스트

- 리스트에서는 페이지를 0으로 받도록 했는데 1로 받는게 나은가요?
- Create 할때 reponse 는 없애려고 합니다

ohsuha committed:
[21] product 단건 조회

f-lab-kai-c commented on src/main/java/org/example/commerce_site/application/address/AddressFacade.java:
```diff
@@ -14,8 +19,30 @@ public class AddressFacade {
 	private final UserService userService;
 
-	public AddressResponseDto.Create create(AddressRequestDto.Create dto) {
+	public void create(AddressRequestDto.Create dto) {
```
return을 주지 않기로 한 이유가 궁금합니다

ohsuha replied:
이전 멘토링받을때 생성과 수정은 return 을 주지 않아도 괜찮다고 하여 없애는 쪽으로 하려 했습니다..!

f-lab-kai-c replied:
return을 주지 않아도 괜찮다와 없앤다에는 살짝 간극이 있는 것 같은데, 왜 없애는 방향으로 진행했는지 궁금합니다.

ohsuha replied:
헉 같은말이 아니였나요...? 어떤 차이가 있는지 잘 모르겠습니다..ㅠ

f-lab-kai-c commented on src/main/java/org/example/commerce_site/common/response/ApiSuccessResponse.java:
```diff
@@ -23,4 +25,31 @@ public static ApiSuccessResponse success() {
 			.build();
 	}
+
+	//for list api
```
이 주석은 필요한 주석인가요?

ohsuha replied:
불필요한 주석인것같네요 삭제하겠습니다

f-lab-kai-c commented on src/main/java/org/example/commerce_site/representation/address/AddressController.java:
```diff
@@ -21,9 +28,42 @@ public class AddressController {
+	@GetMapping("/{user_id}/list")
+	public ApiSuccessResponse.PageList<AddressResponse.Get> getAddresses(
+		@PathVariable(name = "user_id") Long userId, //TODO 추후 토큰에서 가져오도록 함
+		@RequestParam(value = "page", defaultValue = "1") int page,
+		@RequestParam(value = "size", defaultValue = "10") int size) {
```
page는 기본적으로 0으로 시작합니다. 추가로 스프링에서는 기본적인 pagination에 대한 지원을 제공하는데, `Pageable` 관련해서 찾아보시면 좋을 것 같습니다.

ohsuha committed:
[25] 코드리뷰 반영

- 페이지를 1로 받도록하고 내부에서 0으로 처리하도록 수정
- 불필요한 주석 삭제

ohsuha committed:
[25] 코드리뷰 반영

- 페이지를 1로 받도록하고 내부에서 0으로 처리하도록 수정
- 불필요한 주석 삭제

ohsuha committed:
[25] flyaway 적용

flyaway 선택이유
1. 명시적 마이그레이션 관리로 언제 어떤 변경이 있었는지 추적 가능
2. 롤백 가능
3. 운영환경이라고 생각하고 개발하고 싶습니다
4. 처음 보는 거라 써보고 싶어서...

ohsuha committed:
[20] updatePrimary 를 매개변수 받도록 수정

- 어떻게 업데이트를 할것인지를 매개변수 통해서 받아 메서드가 어떻게 동작하는지 명확히하고자함

ohsuha committed:
[20] transactional 어노테이션 추가

- readOnly = true 시 변경 감지를 위한 추가 작업(더티 체크)을 생략, 읽기 작업 수행시 실수로 데이터 수정하는 것 방지, 롤백을 고려할 필요 없어 커넥션이 효율적으로 관리됨의 이점에 따라 정보를 읽어오는 서비스 로직에 적용했습니다.

ohsuha committed:
[21] product 생성시 isDeleted false 로 추가

ohsuha committed:
[21] product list

- queryDSL을 사용해 카테고리명과 상품등록자(파트너)명을 가져오도록 했습니다.
- 상품 이름으로 검색이 가능하게 했습니다.

ohsuha committed:
Merge pull request #28 from f-lab-edu/feature/product/21-product-list

[21] 상품 리스트 조회

ohsuha committed:
Merge branch 'feature/20-product-modify-delete' into feature/user/25-address-update-del-list

ohsuha committed:
[25] 주소 업데이트 기능 삭제

- 주소 변경시 주소를 먼저 등록 후 변경하도록 수정
